app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -24,11 +24,47 @@ const TabCommonPropTypes = {
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
-const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
+let tabTypeComponent = []
+
+function isThisTabType(tab,tabType) {
+  return tab.type === tabType;
+}
+
+const isObjectEmpty = (objectName) => {
+  return (
+      objectName &&
+      Object.keys(objectName).length === 0 &&
+      objectName.constructor === Object
+  );
+};
+
+function enableExperimentPageTab(tab) {
```

Comment:
@upendrakumbham I am going to wait where are you going with this as @lingyun1010 also pointed out that this code is not doing what we want.
From me a couple of things:
- this function is doing a side effect (modifying the `tabTypeComponent`), so doing at least 2 things
- all the complex code inside every `if` condition could be extracted to a well named `validate...` function
- the name of the function is `enableExperimentPageTab` but it is returning a String with a tab name, so nothing to do with is name TBH
Could you consider these comments, please? Thanks